### ISSUE TITLE
Fails during Diarization

### ISSUE THREAD

**Issue by DanOnline2023** [thomasvvugt/whisperx#1]
Hello. Thanks so much for this Docker image. It works perfectly with no diarization.

I understand that you are not the original author of WisperX, but I was hoping you could shed some light on failure during Diarization. I'm attaching a screen shot showing the error.

Thanks so much,

Dan
![Diarization Failure](https://github.com/user-attachments/assets/69b937b7-2229-4ae0-866d-a4673a63b034)

**Comment by thomasvvugt** [comment]
Anytime - speak diarization requires an authentication token from the Hugging face platform as described here: [https://github.com/m-bain/whisperX#speaker-diarization](https://github.com/m-bain/whisperX#speaker-diarization)

Let me know if you need any help :)

**Comment by DanOnline2023** [comment]
Hi. Thanks for getting back.

If you look at the Docker run command in the first line of the screenshot I provided, you will see that I am providing a HF token via the --hf_token command line switch. The token has been blurred to comply with HF's terms of service.

There must be something else going on here.

Thanks so much for your help,

Dan

**Comment by thomasvvugt** [comment]
Hi, sorry for the delay on my end.

I tried recreating the issue you're facing with diarization an english recording on a CPU, but I could not reproduce this behaviour on my end; neither with the CUDA image.

Looks to me like the pyannote/speaker-diarization-3.1 could not be downloaded from HF.

KR, Thomas

**Comment by thomasvvugt** [comment]
Did you accept their user agreement as stated in the error message?

I pushed some new images with the latest whisperx codebase, maybe that helps solving your issue.
Might introduce automated GitHub Actions and better tagging in the future.

**Comment by DanOnline2023** [comment]
I found the problem. I had accepted the user agreement for pyannote/segmentation-3.1, but you're using 3.0. Once I accepted the user agreement for 3.0, it worked immediately. Also, I pulled your latest Docker image. Thanks for the update.

The diarization is not perfect, but it's plenty good enough for my purposes.

Thanks so much for sharing your image on Docker Hub, and for your help here.

Dan
